File: win-prog/ProgWinCS/Text and Fonts/InstalledFontsList/InstalledFontsList.cs
//-------------------------------------------------
// InstalledFontsList.cs © 2001 by Charles Petzold
//-------------------------------------------------
using System;
using System.Drawing;
using System.Drawing.Text;
using System.Windows.Forms;

class InstalledFontsList: FamiliesList
{
     public new static void Main()
     {
          Application.Run(new InstalledFontsList());
     }
     public InstalledFontsList()
     {
          Text = "InstalledFontCollection List";
     }
     protected override FontFamily[] GetFontFamilyArray(Graphics grfx)
     {
          FontCollection fc = new InstalledFontCollection();
          return fc.Families;
     }
}

File: win-prog/ProgWinVB/Text and Fonts/InstalledFontsList/InstalledFontsList.vb
'---------------------------------------------------
' InstalledFontsList.vb (c) 2002 by Charles Petzold
'---------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Text
Imports System.Windows.Forms

Class InstalledFontsList
    Inherits FamiliesList

    Shared Shadows Sub Main()
        Application.Run(New InstalledFontsList())
    End Sub

    Sub New()
        Text = "InstalledFontCollection List"
    End Sub

    Protected Overrides Function GetFontFamilyArray _
                                (ByVal grfx As Graphics) As FontFamily()
        Dim fc As New InstalledFontCollection()
        Return fc.Families
    End Function
End Class
